api/model/storage/postgres/dataset.go

```diff
@@ -216,20 +216,40 @@ func (s *Storage) parseData(rows pgx.Rows) ([][]string, error) {
 }
 
 // FetchDataset extracts the complete raw data from the database.
-func (s *Storage) FetchDataset(dataset string, storageName string) ([][]string, error) {
+func (s *Storage) FetchDataset(dataset string, storageName string, invert bool, filterParams ...*api.FilterParams) ([][]string, error) {
 	// get data variables (to exclude metadata variables)
 	vars, err := s.metadata.FetchVariables(dataset, true, false)
 	if err != nil {
 		return nil, err
 	}
-
-	varNames := []string{}
+	filteredVars := []*model.Variable{}
+	// only include data with distilrole data and index
 	for _, v := range vars {
+		check := v.DistilRole == model.VarDistilRoleIndex || v.DistilRole == model.VarDistilRoleData
```

Comment:
https://github.com/uncharted-distil/distil-compute/blob/master/model/model.go#L521

Should probably use this function instead